src/main/java/core/basesyntax/serviceimpl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator<List<String>> {
+    @Override
+    public List<String> getReport() {
+        Map<String, Integer> fruitStorage = Storage.fruitStorage;
```

Comment:
The use of a public static variable for `fruitStorage` is not recommended. It exposes the internal representation of the storage, which breaks encapsulation. Consider providing access to this storage via instance methods.